src/lib/core/exception.cc

```diff
@@ -172,6 +174,15 @@ SocketError::SocketError(const char *file, unsigned line,
 	errno = save_errno;
 }
 
+const struct type_info type_GaiError =
+	make_type("GaiError", &type_GaiError);
```

Comment:
You didn't add `GaiError` support to `mp_error`. Its type `"GaiError"` is lost when marshalled through network. Maybe you could reuse `SystemError` here. Perhaps inline what you did in the constructor? Also could keep errno as `EIO` and change only the error message. Errno does not seem to be relevant to the problem.